compiler/noirc_frontend/src/elaborator/mod.rs

```diff
@@ -1302,6 +1302,7 @@ impl<'context> Elaborator<'context> {
 
         for (local_module, id, func) in &mut function_set.functions {
             self.local_module = *local_module;
+            self.in_contract = self.module_id().module(self.def_maps).is_contract;
```

Comment:
We should set `in_contract` back to false or its previous value after the call as well